### PR TITLE
Add error messages for failed requests

### LINKED ISSUE #631: Error Snack does not work

### PR DESCRIPTION
**Pull request by jefferymills**

### CODE REVIEW

File: src/slices/appSlice.ts [hunk 1]
```diff
@@ -61,6 +72,16 @@ export const appSlice = createSlice({
...
+  extraReducers: (builder) => {
+    builder.addMatcher(configAPI.endpoints.getConfig.matchRejected, (state, action) => apiErrorHandler(state, action));
+    builder.addMatcher(displayInfoAPI.endpoints.getDisplayInfo.matchRejected, (state, action) =>
+      apiErrorHandler(state, action),
+    );
+    builder.addMatcher(metaDataAPI.endpoints.getMetaData.matchRejected, (state, action) => apiErrorHandler(state, action));
+    builder.addMatcher(displayListAPI.endpoints.getDisplayList.matchRejected, (state, action) =>
+      apiErrorHandler(state, action),
+    );
+  },
```
Comment (jefferymills) [inline src/slices/appSlice.ts:78]: This is the main gist of how it works.

File: src/selectors/app.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import type { RootState } from '../store';
+
+export const selectAppId = (state: RootState) => state.app.appId;
+export const selectBasePath = (state: RootState) => state.app.basePath;
+export const selectConfigPath = (state: RootState) => state.app.configPath;
+export const selectDialogOpen = (state: RootState) => state.app.dialog;
+export const selectErrorMessage = (state: RootState) => state.app.errorMsg;
```
Comment (jefferymills) [inline src/selectors/app.ts:7]: Had to move selectors to their own file to avoid circular dependency.

**Comment by Incin** [comment]
This all looks good but in the old app, the error snack handled like 10 different errors that were thrown. Will this account for all of those cases?

**Review by Incin — APPROVED** [review]

**Comment by jefferymills** [comment]
> This all looks good but in the old app, the error snack handled like 10 different errors that were thrown. Will this account for all of those cases?

I only see uses of `setErrorMessage` inside `actions/index.ts`. Are there others what we've removed?

**Comment by dpenciso** [comment]
I just went back and looked at how it's handled on master/develop and it looks like there are different messages depending under which conditions it fails. Looks like there are 9 different messages. I guess it just depends on how specific we want to be with our messaging on an error.

**Pull request merged**